File: dataStructure/selectionSort/selection_sort.py
# Ordenamiento por selección


def selection_sort(list_of_numbers: list) -> list:
    """
    Sorts a list of numbers in ascending order using the selection sort algorithm.
    The objective of the selection_sort function is to sort a given list of numbers in ascending order using the selection sort algorithm.

    Args:
        list_of_numbers (list): The list of numbers to be sorted.

    Returns:
        list: The sorted list of numbers.
    """
    for i in range(len(list_of_numbers)):
        minimum = i
        for j in range(i+1, len(list_of_numbers)):
            if (list_of_numbers[j] < list_of_numbers[minimum]):
                minimum = j
        temp = list_of_numbers[i]
        list_of_numbers[i] = list_of_numbers[minimum]
        list_of_numbers[minimum] = temp
        
    return list_of_numbers


print(selection_sort([5,4,3,9]))
print(selection_sort([1,2,3,4,5]))
print(selection_sort([10,9,8,7,6]))

File: dataStructure/selectionSort/selectionSort.js
// Método de Selección

const selectionSort = (arrayOfNumbers = []) => {
  const length = arrayOfNumbers.length;
  for (let index = 0; index < length; index++) {
    let min = index;
    for (let j = index + 1; j < length; j++) {
      if (arrayOfNumbers[j] < arrayOfNumbers[min]) {
        min = j;
      }
    }
    const temp = arrayOfNumbers[index];
    arrayOfNumbers[index] = arrayOfNumbers[min];
    arrayOfNumbers[min] = temp;
  }
  return arrayOfNumbers;
};

console.log(selectionSort([8, 10, 6, 7, 5]));
console.log(selectionSort([5, 2, 8, 3, 1]));
